src/shell/commands/data_operations.cpp

```diff
@@ -2201,6 +2212,42 @@ bool count_data(command_executor *e, shell_context *sc, arguments args)
         }
     }
 
+    if (!precise) {
+        if (need_scan) {
+            fprintf(stderr,
```

Comment:
So if the user specifies `--partition`, she must specify `--precise` too?

Reply:
Yeah, just for simplicity. For `estimate count` can be get  immediately, it's not necessary to filter `partition`